backend/static-db/src/users/entities/user.entity.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  BeforeInsert,
+  BaseEntity,
+  TableInheritance,
+} from 'typeorm';
+import { IsEmail } from 'class-validator';
+import { Logger } from '@nestjs/common';
+import { jwtConstants } from '../../../config/auth/jwt.secret';
+import * as jwt from 'jsonwebtoken';
+import * as bcrypt from 'bcryptjs';
+
+@Entity('User')
+@TableInheritance({ column: { type: 'varchar', name: 'type' } })
+export class User {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ type: 'text' })
+  firstName: string;
+
+  @Column({ type: 'text' })
+  lastName: string;
+
+  @Column({ unique: true })
+  @IsEmail()
+  email: string;
+
+  @Column({ unique: true, nullable: false })
+  phoneNumber: string;
+
+  @CreateDateColumn({ type: 'timestamp' })
+  createdAt: Date;
+
+  @Column({ nullable: false })
+  password: string;
+
+  @Column()
+  accountType: string;
+
+  @Column()
+  salt: string;
```

Comment:
Why is there a salt in the database as a column?

Reply:
could it be stored in a different table?

Reply:
The salt is used to compare the decrypted account password with the password that a user is sending when trying to logging into that account. Also, if for some reason the password gets leaked, it can't be decrypted without that salt.

Reply:
> For that reason, It can't be store in a different table as it belongs to the user and the user's password.
Yeah, that makes sense, my original question was just to double-check that having the salt with the user in the same table wouldn't compromise security.